pkg/idtools/idtools.go

```diff
@@ -0,0 +1,145 @@
+package idtools
+
+import (
+	"fmt"
+	"math"
+	"os"
+
+	"github.com/docker/docker/pkg/system"
+)
+
+// IDMap contains a single entry for user namespace range remapping. An array
+// of IDMap entries represents the structure that will be provided to the Linux
+// kernel for creating a user namespace.
+type IDMap struct {
+	ContainerID int `json:"container_id"`
```

Comment:
It has to match **libcontainer** `config.IDMap` for translation between the two.  Without the same json tags, Go-lang refuses to do type assignment between them. (Longer story on why we don't directly use libcontainer's type def'n here.. I can point back to the discussion or we can chat on IRC)